src/mahoji/lib/abstracted_commands/ironmanCommand.ts

```diff
@@ -0,0 +1,120 @@
+				'Would you like to change your ironman to a *permanent* iron? The only thing in your account that will change, is that you will no longer be able to de-iron. This is *permanent* and cannot be reversed.'
+			);
+			await mahojiUserSettingsUpdate(user.id, {
+				bitfield: {
+					push: BitField.PermanentIronman
+				}
+			});
+			return 'You are now a **permanent** Ironman, Enjoy!';
+		}
+		await handleMahojiConfirmation(
+			interaction,
+			'Would you like to stop being an ironman? You will keep all your items and stats but you will have to start over if you want to play as an ironman again.'
+		);
+		await mahojiUserSettingsUpdate(user.id, {
+			minion_ironman: false
+		});
+		return 'You are no longer an ironman.';
+	}
+
+	const existingGiveaways = await prisma.giveaway.findMany({
+		where: {
+			user_id: user.id,
+			completed: false
+		}
+	});
+
+	if (existingGiveaways.length !== 0) {
+		return "You can't become an ironman because you have active giveaways.";
+	}
+
+	await handleMahojiConfirmation(
+		interaction,
+		`Are you sure you want to start over and play as an ironman?
+
+:warning: **Read the following text before confirming. This is your only warning. ** :warning:
+
+The following things will be COMPLETELY reset/wiped from your account, with no chance of being recovered: Your entire bank, collection log, GP/Coins, QP/Quest Points, Clue Scores, Monster Scores, all XP. If you type \`confirm\`, they will all be wiped.
+
+After becoming an ironman:
+	- You will no longer be able to receive GP from  \`+daily\`
+	- You will no longer be able to use \`+pay\`, \`+duel\`, \`+sellto\`, \`+sell\`, \`+dice\`
+	- You can de-iron at any time, and keep all your stuff acquired while playing as an ironman.`
+	);
+
+	const mUser = await mahojiUsersSettingsFetch(user.id);
+
+	type KeysThatArentReset =
+		| 'ironman_alts'
+		| 'main_account'
+		| 'bank_bg_hex'
+		| 'bank_sort_weightings'
+		| 'bank_sort_method'
+		| 'premium_balance_expiry_date'
+		| 'premium_balance_tier'
+		| 'minion_bought_date'
+		| 'id'
+		| 'pets'
+		| 'RSN'
+		| 'patreon_id'
+		| 'github_id';
+	const createOptions: Required<Pick<Prisma.UserCreateInput, KeysThatArentReset>> = {
+		id: user.id,
+		main_account: mUser.main_account,
+		ironman_alts: mUser.ironman_alts,
+		bank_bg_hex: mUser.bank_bg_hex,
+		bank_sort_method: mUser.bank_sort_method,
+		bank_sort_weightings: mUser.bank_sort_weightings as ItemBank,
+		minion_bought_date: mUser.minion_bought_date,
+		github_id: mUser.github_id,
+		patreon_id: mUser.patreon_id,
+		RSN: mUser.RSN,
+		premium_balance_expiry_date: mUser.premium_balance_expiry_date,
+		premium_balance_tier: mUser.premium_balance_tier,
+		pets: mUser.pets as ItemBank
+	};
+
+	try {
+		await prisma.user.delete({
+			where: { id: user.id }
+		});
+		await prisma.user.create({
+			data: createOptions
+		});
+		await prisma.slayerTask.deleteMany({ where: { user_id: user.id } });
+		await prisma.playerOwnedHouse.delete({ where: { user_id: user.id } });
```

Comment:
Isn't there a bug where this will fail if the user doesn't have a PoH? I thought I changed this to add a `.catch(noOp)` afterwards? 

Same with minigame + newUser (Anything using delete and not deleteMany).

Without this, after the first failure, it will refuse to continue to delete the remaining rows, inhibiting a  full wipe.